src/sentry/tsdb/snuba.py

```diff
@@ -587,6 +589,9 @@ def __get_data_legacy(
             orderby.append(model_group)
 
         if keys:
+            referrer = f"tsdb-modelid:{model.value}"
+            tenant_ids = tenant_ids or dict()
+            tenant_ids["referrer"] = referrer
```

Comment:
Since referrer is already passed in to the snuba call, you could add referrer automatically to the tenant IDs in that code. Then you wouldn't need as many changes in Sentry code.